### DISCUSSION TITLE
Workaround needed for Dynatrace OneAgent for NodeJS

### DISCUSSION THREAD

**Discussion by alumni** [renovatebot/renovate#23348] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

[Dynatrace OneAgent](https://www.npmjs.com/package/@dynatrace/oneagent) is using semver prerelease syntax for both stable and unstable releases (e.g. `<major>.<minor>.<patch>-1.0.0`).

| `package.json` | `latest` | `next` | 
| - | - | - |
| `1.265.152-1.0.0` | `1.267.125-1.0.0` | `1.269.154-1.0.0` |

My current workaround is to follow the `latest` tag:
```json
{
    "description": "Dynatrace OneAgent upgrades",
    "matchPackageNames": ["@dynatrace/oneagent"],
    "followTag": "latest"
}
```

Another option would be to treat the part after `-` as `build`/`revision`, however this will match the unstable release:
```json
{
    "description": "Dynatrace OneAgent upgrades",
    "matchPackageNames": ["@dynatrace/oneagent"],
    "versioning": "regex:^(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?(-(?<build>.*))?$"
}
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
I think following latest tag makes most sense. Their versioning approach sounds like a very painful idea

**Reply by alumni** [reply]
I agree. I opened the thread in case you would like to include the workaround :)
